File: ProjetAttractionSpring/ProjetAttractionBack/src/main/java/attraction/model/Achat.java
package attraction.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class Achat {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	@ManyToOne
	@JoinColumn(name="id_element")
	private Element element;
	private int niveauAmelioration;
	private int nbSameElement;
	@Column(name="type_element",length = 40)
	private String typeElement;
	@ManyToOne
	@JoinColumn(name="id_parc")
	private Parc parc;


	public Achat(Integer id, Element element,int niveauAmelioration,String typeElement,Parc parc) {
		this.id = id;
		this.element = element;
		this.niveauAmelioration = niveauAmelioration;
		this.typeElement=typeElement;
		this.parc=parc;
	}

	public Achat(Element element,int niveauAmelioration,String typeElement,Parc parc) {
		this.element = element;
		this.niveauAmelioration = niveauAmelioration;
		this.typeElement=typeElement;
		this.parc=parc;
	}

	public Achat(Element element,int nbSameElement,int niveauAmelioration,String typeElement,Parc parc) {
		this.element = element;
		this.niveauAmelioration = niveauAmelioration;
		this.nbSameElement=nbSameElement;
		this.typeElement=typeElement;
		this.parc=parc;
	}

	public Achat() {}

	
	

	public String getTypeElement() {
		return typeElement;
	}

	public void setTypeElement(String typeElement) {
		this.typeElement = typeElement;
	}
	
	public int getNbSameElement() {
		return nbSameElement;
	}

	public void setNbSameElement(int nbSameElement) {
		this.nbSameElement = nbSameElement;
	}

	public int getNiveauAmelioration() {
		return niveauAmelioration;
	}

	public Integer getId() {
		return id;
	}


	public void setId(Integer id) {
		this.id = id;
	}


	public Element getElement() {
		return element;
	}


	public void setElement(Element element) {
		this.element = element;
	}





	public void setNiveauAmelioration(int niveauAmelioration) {
		this.niveauAmelioration = niveauAmelioration;
	}

}

File: ProjetAttractionAngular/src/app/model/achat.ts
import { Element } from './element';
import { Parc } from './parc';

export class Achat {

  	constructor(
	  private _parc : Parc,
	  private _element : Element,
	  private _typeElement : string,
	  private _niveauAmelioration : number,
	  private _nbSameElement : number,
    	private _id? : number
 	){}


	public get typeElement() : string {
		return this._typeElement;
	}

	public set typeElement(typeElement : string) {
		this._typeElement = typeElement;
	}

	public get nbSameElement() : number {
		return this._nbSameElement;
	}

	public set nbSameElement(nbSameElement : number) {
		this._nbSameElement = nbSameElement;
	}

	public get niveauAmelioration() : number {
		return this._niveauAmelioration;
	}

	public set niveauAmelioration(niveauAmelioration : number) {
		this._niveauAmelioration = niveauAmelioration;
	}

	public get id() : number | undefined {
    return this._id;
	}


	public set id(id : number|undefined) {
		this._id = id;
	}


	public get element() : Element {
		return this._element;
	}


	public set element(element : Element) {
		this._element = element;
	}


	public get parc() : Parc {
		return this._parc;
	}


	public set parc(parc : Parc) {
		this._parc = parc;
	}

}
